src/test/java/de/qabel/desktop/crashReports/HockeyAppTest.java

```diff
@@ -0,0 +1,174 @@
+import de.qabel.core.accounting.CloseableHttpResponseStub;
+import de.qabel.desktop.hockeyapp.HockeyAppConfiguration;
+import de.qabel.desktop.hockeyapp.HockeyAppVersion;
+import de.qabel.desktop.hockeyapp.VersionClient;
+import org.apache.http.NameValuePair;
+import org.apache.http.entity.BasicHttpEntity;
+import org.jetbrains.annotations.NotNull;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+public class HockeyAppTest {
+
+    private final String shortVersion = "1.1";
+
+    CloseableHttpClientStub httpClient = new CloseableHttpClientStub();
+
+    HockeyAppConfiguration config = new HockeyAppConfiguration(shortVersion, httpClient);
+    VersionClient versionClient = new VersionClient(config, httpClient);
+    HockeyApp hockeyApp = new HockeyApp(shortVersion, httpClient);
+
+    private String feedback;
+    private String name;
+    private String email;
+    private String stacktrace;
+
+    public HockeyAppTest() {
+        versionClient.setVersion(new HockeyAppVersion(1, config.getAppVersion()));
+    }
+
+    @Before
+    public void setUp() throws Exception {
+        feedback = "HockeyAppTest feedback text";
+        name = "HockeyAppTest";
+        email = "HockeyAppTest@desktop.de";
+        stacktrace = "XCEPTION REASON STRING\n" +
+            "  at CLASS.METHOD(FILE:LINE)\n" +
+            "  at CLASS.METHOD(FILE:LINE)\n" +
+            "  at CLASS.METHOD(FILE:LINE)\n" +
+            "  at CLASS.METHOD(FILE:LINE)\n" +
+            "  at CLASS.METHOD(FILE:LINE)\n" +
+            "  at CLASS.METHOD(FILE:LINE)\n" +
+            "  at CLASS.METHOD(FILE:LINE)\n" +
+            "  at CLASS.METHOD(FILE:LINE)\n" +
+            "  at CLASS.METHOD(FILE:LINE)\n" +
+            "ANOTHER EXCEPTION REASON STRING\n" +
+            "  at CLASS.METHOD(FILE:LINE)\n" +
+            "  at CLASS.METHOD(FILE:LINE)\n" +
+            "  at CLASS.METHOD(FILE:LINE)\n" +
+            "  at CLASS.METHOD(FILE:LINE)";
+
+        stubFeedbackResponse();
+        stubVersionToApp();
+    }
+
+
+    @Test
+    public void testSendFeedback() throws IOException {
+
+        hockeyApp.sendFeedback(feedback, name, email);
+        String responseContent = getStubFeedbackResponseContent();
+        JSONObject json = parseFeedbackJson(responseContent);
+
+        assertEquals(123, json.get("id"));
+        assertEquals(shortVersion, versionClient.getVersion().getShortVersion());
+    }
+
+
+    @Test
+    public void testSendCrashReport() throws IOException {
+        stubSendCrashReport();
+        hockeyApp.sendStacktrace(feedback, stacktrace);
+    }
+
+    @Test
+    public void createLog() throws IOException {
+        String containingString = "Version: 1.1";
+        stacktrace = hockeyApp.createLog(stacktrace);
```

Comment:
You saved a variable but it gets confusing. Just create a new variable with a meaningful name like `formattedStacktrace` or `hockeyappFriendlyStacktrace` or whatever you like.